From: David Decotigny <decot@google.com>
Date: Wed, 27 Apr 2011 18:32:39 +0000
Subject: [PATCH 58/71] ethtool: Use full 32 bit speed range in ethtool's
 set_settings

commit 25db0338813a8915457636b1f6abe6a28fa73f8d upstream.

This makes sure the ethtool's set_settings() callback of network
drivers don't ignore the 16 most significant bits when ethtool calls
their set_settings().

All drivers compiled with make allyesconfig on x86_64 have been
updated.

Signed-off-by: David Decotigny <decot@google.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
[bwh: Restricted to drivers/net/r8169.c]
---
 drivers/net/r8169.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/drivers/net/r8169.c b/drivers/net/r8169.c
index 64c5ea0..77370c0 100644
--- a/drivers/net/r8169.c
+++ b/drivers/net/r8169.c
@@ -1352,7 +1352,8 @@ static int rtl8169_set_settings(struct net_device *dev, struct ethtool_cmd *cmd)
 
 	spin_lock_irqsave(&tp->lock, flags);
 	ret = rtl8169_set_speed(dev,
-		cmd->autoneg, cmd->speed, cmd->duplex, cmd->advertising);
+				cmd->autoneg, ethtool_cmd_speed(cmd),
+				cmd->duplex, cmd->advertising);
 	spin_unlock_irqrestore(&tp->lock, flags);
 
 	return ret;
-- 
1.7.5.4

